### PR TITLE
Create documentation on deployment CI/CD process

### PR DESCRIPTION
**Pull request by maestromac**
## What type of PR is this? (check all applicable)
- [x] Documentation Update

## Description
Create documentation on deployment CI/CD process. Deployment portion is written by @mstruve 
## Related Tickets & Documents
Resolve https://github.com/thepracticaldev/dev.to/issues/4494
## Mobile & Desktop Screenshots/Recordings (if there are UI changes)
n/a
## Added to documentation?
- [x] no documentation needed

### CODE REVIEW

File: docs/deployment.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+title: Deployment and CI/CD Process
+---
+
+# Deployment and CI/CD Process
+
+## Overview
+
+DEV relies on GitHub + Travis to continuous deploy to Heroku. If a Pull Request is merged without the "ci skip" in it's title, it will be automatically deployed to production once the test suite passes. The process currently takes about 20 minute to complete and will need a few addtional minute before the change live.
```
Comment (mstruve) [inline docs/deployment.md]: "minutes to complete"
"minutes before the change"

File: docs/deployment.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+The following steps can be explored in our [.travis.yml](https://github.com/thepracticaldev/dev.to/blob/master/.travis.yml) and [Procfile](https://github.com/thepracticaldev/dev.to/blob/master/Procfile)
+
+1. Travis runs the test portion of Rails code
+1. Travis runs the test portion of Preact code
+1. CodeClimate-test-reporter combines the test result and coverage from Ruby and JavaScript code then upload it to our CodeClimate dashboard.
+1. `bundle-audit` check for any known vulnerability
+1. Travis builds Storybook to ensure it's integrity.
+1. Travis deploys code to Heroku
+   - Heroku runs migration krior to deployment
```
Comment (mstruve) [inline docs/deployment.md]: "prior to deployment"

File: docs/deployment.md [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+1. `after_deploy` scripts kicks in
+   - Airbrake Deploy Tracking is notified of the doployment
+1. Travis notifies the team the process completed
+
+## Deploying to Heroku
+
+We use Heroku's Release Phase feature.
```
Comment (mstruve) [inline docs/deployment.md]: Should we add a link here to the [Release Phase docs](https://devcenter.heroku.com/articles/release-phase)?

File: docs/deployment.md [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+## Overview
+
+DEV relies on GitHub + Travis to continuous deploy to Heroku. If a Pull Request is merged without the "ci skip" in it's title, it will be automatically deployed to production once the test suite passes. The process currently takes about 20 minute to complete and will need a few addtional minute before the change live.
+
+## Travis steps
+
+The following steps can be explored in our [.travis.yml](https://github.com/thepracticaldev/dev.to/blob/master/.travis.yml) and [Procfile](https://github.com/thepracticaldev/dev.to/blob/master/Procfile)
+
+1. Travis runs the test portion of Rails code
+1. Travis runs the test portion of Preact code
```
Comment (mstruve) [inline docs/deployment.md]: Do you want these all to be "1." or increment? Figured some might run in parallel but the final steps I know are synchronous.

File: docs/deployment.md (hunk 1 above)
Comment (rhymes) [inline docs/deployment.md]: `DEV relies on GitHub + Travis to continuous deploy to Heroku` -> `DEV relies on GitHub and Travis to deploy continuously to Heroku`
`in it's title` -> `in its title`
`once the test suite passes` -> `once the build steps complete succesfully` (after all it's not just the tests)
`20 minute` -> `20 minutes`
`addtional minute` -> `additional minutes`
`before the change live` -> `before the change goes live`

File: docs/deployment.md [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+1. CodeClimate-test-reporter combines the test result and coverage from Ruby and JavaScript code then upload it to our CodeClimate dashboard.
```
Comment (rhymes) [inline docs/deployment.md]: `upload it` -> `uploads it`

I'd remove the full stop at the end as well

File: docs/deployment.md [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+1. `bundle-audit` check for any known vulnerability
```
Comment (rhymes) [inline docs/deployment.md]: `check` -> `checks`

File: docs/deployment.md [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+1. Travis builds Storybook to ensure it's integrity.
```
Comment (rhymes) [inline docs/deployment.md]: `it's integrity` -> `its integrity` and same thing about the full stop

ps. aside: do we actually use Storybook?

File: docs/deployment.md (hunk 2 above)
Comment (rhymes) [reply docs/deployment.md]: `migration` -> `the database migrations`

File: docs/deployment.md [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+1. Travis deploys code to Heroku
+   - Heroku runs migration krior to deployment
+1. `after_deploy` scripts kicks in
+   - Airbrake Deploy Tracking is notified of the doployment
```
Comment (rhymes) [inline docs/deployment.md]: `doployment` -> `deployment`

File: docs/deployment.md [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+1. Travis notifies the team the process completed
```
Comment (rhymes) [inline docs/deployment.md]: `that the process is completed`

File: docs/deployment.md (hunk 4 above)
Comment (rhymes) [reply docs/deployment.md]: I think we could run these in parallel in theory

**Comment by maestromac** [comment]
This is as rough as a rough draft gets @rhymes :sweat_smile: Thanks for the review

File: docs/deployment.md (hunk 7 above)
Comment (maestromac) [reply docs/deployment.md]: No, we haven't been using it.

File: docs/deployment.md (hunk 1 above)
Comment (maestromac) [reply docs/deployment.md]: Updated!

File: docs/deployment.md (hunk 4 above)
Comment (maestromac) [reply docs/deployment.md]: The all "1"s is an acceptable Markdown Syntax. You can view [the file itself](https://github.com/thepracticaldev/dev.to/blob/848d3aedddf498ab36aed50f6500a96be8c746d0/docs/deployment.md) to see that the numbers are properly incremented.

Comment (mstruve) [reply docs/deployment.md]: TIL

File: docs/deployment.md [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+
+# Deployment and CI/CD Process
+
+## Overview
+
+DEV relies on GitHub and Travis to deploy continuously to Heroku. If a Pull Request is merged without the "ci skip" in its title, it will be automatically deployed to production once the build steps complete successfully. The process currently takes about 20 minutes to complete and will need a few additional minutes before the change goes live.
+
+## Travis steps
+
+The following steps can be explored in our [.travis.yml](https://github.com/thepracticaldev/dev.to/blob/master/.travis.yml) and [Procfile](https://github.com/thepracticaldev/dev.to/blob/master/Procfile). Some of the steps will be parallelized in the future.
```
Comment (rhymes) [inline docs/deployment.md]: maybe a colon at the end of future? since it's introducing the list i mean

Comment (maestromac) [reply docs/deployment.md]: Updated!

File: docs/deployment.md [hunk 11]
```diff
@@ -0,0 +1,30 @@
+---
+title: Deployment and CI/CD Process
+---
+
+# Deployment and CI/CD Process
+
+## Overview
+
+DEV relies on GitHub and Travis to deploy continuously to Heroku. If a Pull Request is merged without the "ci skip" in its title, it will be automatically deployed to production once the build steps complete successfully. The process currently takes about 20 minutes to complete and will need a few additional minutes before the change goes live.
```
Comment (mstruve) [inline docs/deployment.md]: Don't forget we need to update this if/when your other PR goes out 💃

Comment (maestromac) [reply docs/deployment.md]: Yup! I'll merge this then update it on the other PR.

**Pull request merged**